hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameBuilder.java

```diff
@@ -155,9 +156,11 @@ private MessageFrame finishedAsCall(
             @NonNull final FeatureFlags featureFlags,
             @NonNull final Configuration config) {
         Code code = CodeV0.EMPTY_CODE;
-        if (canLoadCodeFromAccount(transaction, worldUpdater)) {
+        final var contractId = transaction.contractIdOrThrow();
+
+        if (canLoadCodeFromAccount(transaction, worldUpdater, contractId, config, featureFlags)) {
             final var account = worldUpdater.getHederaAccount(to);
-            if (account == null && worldUpdater.contractMustBePresent()) {
+            if (account == null && contractMustBePresent(config, featureFlags, contractId)) {
```

Comment:
i'm losing track - how is `contractMustBePresent` different from `contractRequired`?  I mean, I see they take different parameters, but given that they compute the same predicate - perhaps they should actually have the same name?  Or would _that_ be more confusing?

Reply:
Really, the reason I needed this is because I don't have a `messageFrame` to work with because I'm in the middle of creating one for the transaction